packages/flutter/lib/src/material/scaffold.dart

```diff
@@ -395,6 +394,16 @@ class _ScaffoldLayout extends MultiChildLayoutDelegate {
       floatingActionButtonRect = fabOffset & fabSize;
     }
 
+    if (hasChild(_ScaffoldSlot.snackBar)) {
+      snackBarSize = layoutChild(_ScaffoldSlot.snackBar, fullWidthConstraints);
```

Comment:
This value is used by the floatingActionButton code above (line 383).

I think you will need to have two possible code paths, one in which we do the FAB then the snackbar, for the new case where the FAB pushes the snackbar, and one in which we do the snackbar then the FAB, for the old case where the FAB moves out of the way of snackbars.

Reply:
That's true, fixed this and added a test since there was no test to confirm FAB position originally.